composer/core/state.py

```diff
@@ -505,9 +505,48 @@ def load_state_dict(self, state: Dict[str, Any], strict: bool = False):
         """
         state = _ensure_backwards_compatible_checkpointing(state)
 
+        # Verify surgery algorithms from checkpoint are also enabled in current run
+        if 'algorithms' in state:
+            import composer.algorithms as algorithms
+
+            # Get the types and counts of existing algorithms
+            existing_algorithm_counts = {}
+            for algorithm in self.algorithms:
+                if algorithm not in existing_algorithm_counts:
+                    existing_algorithm_counts[type(algorithm)] = 0
+                existing_algorithm_counts[type(algorithm)] += 1
+
+            # Get the types and counts of checkpoint algorithms
+            checkpoint_algorithm_counts = {}
+            for algorithm in state['algorithms'].keys():
```

Comment:
similiar to above, can simply via:

```
checkpoint_algo_counts = defaultdict(int)

for algo in state['algorithms'].keys():
    try:
        if getattr(algorithms, algo).is_model_surgery:
            checkpoint_algo_counts[algo] += 1
     except AttributeError:
         # do warning and pass

```

Reply:
+1 to default dict, but want to push back on `try except`. It's a little longer code, but that's ok imo. I'd rather not rely on try catch like this just to skip an additional check

Reply:
that's fine, its a EAFP preference to be explicit.